### PR TITLE
Allow defining custom prices in checkout mutations

### PR DESCRIPTION
**Pull request by IKarbowiak**
- Add `HANDLE_CHECKOUTS` permission.
  - The new permission is granted to all apps and groups that already have `MANAGE_CHECKOUT`.
- The `CheckoutLineInput` is extended with the optional price field `price: PositiveDecimal`, so the custom price could be set in the following mutations:
  - `checkoutCreate`
  - `checkoutLinesAdd` - when adding a variant that is already added to checkout, the already existing line is overridden - the quantity is incremented and the price is updated.
  - `checkoutLinesUpdate` - overrides the existing line with the price provided in the mutation

⚠️ 
This is PR to the task branch.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-6191 Allow defining checkout line custom price in checkout mutations](https://app.clickup.com/t/2549495/SALEOR-6191)

**Comment by patrys** [comment]
Task linked: [SALEOR-6089 Custom prices when adding items to checkout](https://app.clickup.com/t/2549495/SALEOR-6089)

File: saleor/checkout/migrations/0040_add_handle_checkouts_permission.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Generated by Django 3.2.12 on 2022-03-08 10:35
+
+from django.core.management.sql import emit_post_migrate_signal
+from django.db import migrations
+
+
+def assing_permissions(apps, schema_editor):
```
Comment (jakubkuc) [inline saleor/checkout/migrations/0040_add_handle_checkouts_permission.py:7]:
```suggestion
def assign_permissions(apps, schema_editor):
```
? ;)

**Review by korycins — COMMENTED** [review]
To add a new permission to App, the Staff user needs to have this perm. Currently, you need to have a super admin, to assign permission `handle_checkout` to staff user, to give him a possibility to create/update such app. Not sure if this is an expected flow.
(`saleor.graphql.app.utils.ensure_can_manage_permissions`)

File: saleor/checkout/migrations/0040_add_handle_checkouts_permission.py (hunk 1 above)
Comment (L3str4nge) [inline saleor/checkout/migrations/0040_add_handle_checkouts_permission.py:7]:
```suggestion
def assign_permissions(apps, schema_editor):
```

**Review by L3str4nge — APPROVED** [review]

File: saleor/graphql/checkout/mutations/checkout_create.py [hunk 2]
```diff
@@ -15,24 +15,35 @@
...
 
 
 class CheckoutLineInput(graphene.InputObjectType):
     quantity = graphene.Int(required=True, description="The number of items purchased.")
     variant_id = graphene.ID(required=True, description="ID of the product variant.")
+    price = PositiveDecimal(
+        required=False,
+        description=(
+            "Custom price of the item. Can be set only by apps "
+            "with handle checkouts permission. When the line with the same variant "
```
Comment (maarcingebala) [inline saleor/graphql/checkout/mutations/checkout_create.py:43]:
```suggestion
            "with `HANDLE_CHECKOUTS` permission. When the line with the same variant "
```

**Review by maarcingebala — APPROVED** [review]

**Review by tomaszszymanski129 — APPROVED** [review]

File: saleor/graphql/checkout/mutations/checkout_create.py [hunk 3]
```diff
@@ -15,24 +15,35 @@
...
 
 if TYPE_CHECKING:
     from ....account.models import Address
+    from .utils import CustomPrice
 
 
 class CheckoutLineInput(graphene.InputObjectType):
     quantity = graphene.Int(required=True, description="The number of items purchased.")
     variant_id = graphene.ID(required=True, description="ID of the product variant.")
+    price = PositiveDecimal(
```
[the remark this answers is not in the document]
Comment (IKarbowiak) [reply saleor/graphql/checkout/mutations/checkout_create.py:39]: IMO, the description is enough.

**Comment by IKarbowiak** [comment]
@korycins I assigned `handle_checkout` also to groups that have `manage_checkouts` permission.

**Review by korycins — APPROVED** [review]

**Pull request merged**